core/src/main/java/org/apache/carbondata/core/datamap/dev/expr/AndDataMapExprWrapper.java

```diff
@@ -59,6 +60,21 @@ public AndDataMapExprWrapper(DataMapExprWrapper left, DataMapExprWrapper right,
     return andBlocklets;
   }
 
+  @Override
+  public List<ExtendedBlocklet> prune(DataMapDistributable distributable,
+                                      List<PartitionSpec> partitionsToPrune)
+          throws IOException {
+    List<ExtendedBlocklet> leftPrune = left.prune(distributable, partitionsToPrune);
+    List<ExtendedBlocklet> rightPrune = right.prune(distributable, partitionsToPrune);
+    List<ExtendedBlocklet> andBlocklets = new ArrayList<>();
+    for (ExtendedBlocklet blocklet : leftPrune) {
+      if (rightPrune.contains(blocklet)) {
```

Comment:
Have you ever validate the correctness of this? The `equals` and `hashCode` method in `ExtendedBlocklet` only make use of its member `segmentId`. I'm not sure whether it is correct.

Reply:
no use this method of the class.
only use the method:

org.apache.carbondata.core.datamap.dev.expr.DataMapExprWrapperImpl#prune(org.apache.carbondata.core.datamap.DataMapDistributable, java.util.List<org.apache.carbondata.core.indexstore.PartitionSpec>)

Reply:
When will you use org.apache.carbondata.core.datamap.dev.expr.AndDataMapExprWrapper#prune(java.util.List<org.apache.carbondata.core.datamap.Segment>, java.util.List<org.apache.carbondata.core.indexstore.PartitionSpec>) or org.apache.carbondata.core.datamap.dev.expr.AndDataMapExprWrapper#prune(org.apache.carbondata.core.datamap.DataMapDistributable, java.util.List<org.apache.carbondata.core.indexstore.PartitionSpec>)?